modules/ROOT/pages/keycloak/authenticating.adoc

```diff
@@ -51,19 +48,180 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
+}
+----
+
+. Allow your application to handle the redirect in `AppDelegate`:
++
+[source,swift]
+----
+func application(_ app: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey: Any] = [:]) -> Bool {
+    do {
+        return try AgsAuth.instance.resumeAuth(url: url as URL)
+    } catch AgsAuth.Errors.serviceNotConfigured {
+        print("AeroGear auth service is not configured")
+    } catch {
+        fatalError("Unexpected error: \(error).")
+    }
+    return false
+}
+----
+****
+
+[role="secondary"]
+.Cordova
+****
+Authentication will be performed on initialization as described xref:keycloak/coding.adoc[here]. To manually redirect to the login screen use:
+[code,javascript]
+----
+authService.login()
+----
+****
+
+[role="secondary"]
+.Xamarin
+****
+. Get an instance of `IAuthService`:
++
+[source,csharp]
+----
+IAuthService service = MobileCore.Instance.GetInstance<IAuthService>();
+----
+
+. Use the `Authenticate` method to start the auth flow:
++
+[source,csharp]
+----
+var authOptions = DependencyService.Get<IAuthenticateOptionsProvider>().GetOptions();
+service.Authenticate(authOptions).ContinueWith(result =>
+{
+    // load some view
+});
+----
+****
+
+== Logging out
+
+The following section describes how to perform a logout.
+
+[role="primary"]
+.Android
+****
+To logout, invoke the `logout` method. This accepts the UserPrincipal that was provided by `handleAuthResult` and has a callback to determine if the logout to the Keycloak or OpenID Connect server was successful.
+
+[source,java]
+----
+
+
+// authService already initialized.
+AuthService authService = MobileCore.getInstance().getService(AuthService.class);
+UserPrincipal currentUser = authService.currentUser();
+
+authService.logout(currentUser, new Callback<UserPrincipal>() {
+    @Override
+    public void onSuccess() {
+        // User Logged Out Successfully and Local Auth Tokens were Deleted
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // An error occurred during logout
+    }
+});
+
 ----
 
-The callback provided in `AuthService#login` will be invoked.
 ****
 
 [role="secondary"]
 .iOS
 ****
+. To logout invoke the `logout` method:
++
+[source,swift]
+----
+do {
+    try AgsAuth.instance.logout(onCompleted: self.onLogoutComplete)
+} catch {
+    fatalError("Error logging out: \(error).")
+}
+----
+
+. Handle any post-logout actions in the callback:
++
+[source,swift]
+----
+func onLogoutComplete(_: Error?) {
+    // post-logout actions
+}
+----
 
 ****
 
 [role="secondary"]
 .Cordova
 ****
 
-****
\ No newline at end of file
+To logout invoke the `logout` function:
+
+[source,javascript]
+----
+authService.logout()
+----
+
+By default, the local tokens obtained during authentication are only deleted when the logout succeeded against the authentication server.
+
+****
+
+[role="secondary"]
+.Xamarin
+****
+To logout invoke the `Logout` method:
+[source,csharp]
+----
+// auth service already initialised
```

Comment:
This does not make much sense.

Reply:
It kind of does for Xamarin because it requires you to initialize the core in a previous step. I have added docs for that to the general dev env setup guide. But i agree that the comment is confusing and i will remove it.